server/src/main/java/org/candlepin/model/Consumer.java

```diff
@@ -893,13 +915,63 @@ public void setContentAccessMode(String contentAccessMode) {
 
     public void setAddOns(Set<String> addOns) {
         this.addOns = addOns;
+        this.updateRHCloudProfileModified();
     }
 
     public void addAddOn(String addOn) {
         this.addOns.add(addOn);
+        this.updateRHCloudProfileModified();
     }
 
     public void removeAddOn(String addOn) {
         this.addOns.remove(addOn);
+        this.updateRHCloudProfileModified();
+    }
+
+    public Date getRHCloudProfileModified() {
+        return this.rhCloudProfileModified;
+    }
+
+    public void setRHCloudProfileModified(Date rhCloudProfileModified) {
+        this.rhCloudProfileModified = rhCloudProfileModified;
+    }
+
+    public void updateRHCloudProfileModified() {
+        this.rhCloudProfileModified = new Date();
+    }
+
+    /**
+     * Check if the consumers facts have changed and contains the cloud profile facts.
+     * It returns true if incoming facts are cloud profile facts
+     *
+     * @param incomingFacts incoming facts
+     * @return a boolean
+     */
+    public boolean checkForCloudProfileFacts(Map<String, String> incomingFacts) {
+
+        if (incomingFacts == null) {
+            return false;
+        }
+
+        if (this.facts != null && incomingFacts != null && this.facts.equals(incomingFacts)) {
+            return false;
+        }
+
+        for (Entry<String, String> entry : incomingFacts.entrySet()) {
+            if (CloudProfileFacts.containsFact(entry.getKey())) {
```

Comment:
This will work, but it will have to iterate over all of the incoming facts (which usually are 200+) and check for each of them if it is a cloud-profile fact. A more performant loop would be to loop over the CloudProfileFacts.values() (which right now are only 11), then check if each of them is included in the incoming facts (and then continue the rest of the checks)

Reply:
Loop is now revised as suggested.